community/kernel/src/main/java/org/neo4j/tooling/GlobalGraphOperations.java

```diff
@@ -155,12 +160,37 @@ protected Relationship fetchNextOrNull()
      */
     public Iterable<RelationshipType> getAllRelationshipTypes()
     {
+        return getAllRelationshipTypes( false );
+    }
+
+    public Iterable<RelationshipType> getAllRelationshipTypesInUse()
+    {
+        return getAllRelationshipTypes( true );
+    }
+
+    private Iterable<RelationshipType> getAllRelationshipTypes( boolean inUse )
+    {
         assertInUnterminatedTransaction();
-        return cast( relationshipTypes.getAllTokens() );
+        Iterable<RelationshipTypeToken> relationshipTypes = this.relationshipTypes.getAllTokens();
+        if ( inUse )
+        {
+            relationshipTypes = filter( new Predicate<Token>()
+            {
+                @Override
+                public boolean test( Token token )
+                {
+                    Statement statement = statementCtxSupplier.get();
```

Comment:
The underlying KeyValueStore has support for doing this type of scan for exactly the reason of being able to get all values of a particular type. The storage format with sorted keys is even such that this type of scan would be efficient. The only tricky bit would be that we would need some way of doing the same type of scan for for keys that have been added since the last rotation. Other than that it is simply a matter of adding a nice API for it.